src/Aspire.Hosting/api/Aspire.Hosting.cs

```diff
@@ -697,6 +796,13 @@ public ContainerRuntimeArgsCallbackContext(System.Collections.Generic.IList<obje
         public System.Threading.CancellationToken CancellationToken { get { throw null; } }
     }
 
+    public sealed partial class CreationScriptAnnotation : IResourceAnnotation
```

Comment:
@sebastienros Why is this in Aspire.Hosting?

Reply:
This can be used by any resource. Other option is to duplicate it in each database integration that needs it. Third option?

Reply:
Duplicate it in each database resource. Probably with a unique name.

Reply:
https://github.com/dotnet/aspire/pull/8535